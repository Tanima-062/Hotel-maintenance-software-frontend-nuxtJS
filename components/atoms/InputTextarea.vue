<template>
  <div class="relative">
    <textarea
      :value="value"
      :placeholder="placeholder"
      :class="`w-full rounded-md border border-solid border-bdColor text-black-dark placeholder-black-light py-2 focus:shadow-outline focus:outline-none ${styles}`"
      :maxlength="maxlength"
      :rows="rows"
      :disabled="isDisabled"
      @input="input"
    />
    <div
      v-if="showTextLength"
      class="absolute right-0 bottom-0 text-inputLength text-sm py-2.5 pr-2"
    >
      {{ inputTextLength }}/ {{ maxlength }}
    </div>
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop, Emit } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: '' })
  placeholder!: string

  @Prop({ default: '' })
  value!: string

  @Prop({ default: 1000 })
  maxlength!: number

  @Prop({ default: 2 })
  rows!: number

  @Prop({ default: false })
  showTextLength!: boolean

  @Prop({ default: 0 })
  inputTextLength!: number

  @Prop({ default: true })
  resize!: boolean

  @Prop({ default: '' })
  height!: string

  @Prop({ default: false })
  isDisabled!: boolean

  get styles(): string {
    const style = []
    if (this.showTextLength) {
      style.push('pr-16')
      style.push('pl-3')
      style.push('resize-none')
    } else {
      style.push('px-3')
    }
    if (!this.resize) {
      style.push('resize-none')
    }
    if (this.height) {
      style.push(this.height)
    }
    return style.join(' ')
  }

  @Emit()
  private input(e: Event & { target: HTMLInputElement }): string {
    return e.target.value
  }
}
</script>
