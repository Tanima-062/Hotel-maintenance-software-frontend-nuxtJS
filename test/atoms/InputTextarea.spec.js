import { shallowMount } from '@vue/test-utils'
import InputTextarea from '@/components/atoms/InputTextarea.vue'

describe('Atoms InputTextarea Check', () => {
  test('InputTextarea Test', () => {
    const wrapper = shallowMount(InputTextarea)

    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
