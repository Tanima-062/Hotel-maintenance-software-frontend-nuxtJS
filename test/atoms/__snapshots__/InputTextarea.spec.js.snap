// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms InputTextarea Check InputTextarea Test 1`] = `
<div
  class="relative"
>
  <textarea
    class="w-full rounded-md border border-solid border-bdColor text-black-dark placeholder-black-light py-2 focus:shadow-outline focus:outline-none px-3"
    maxlength="1000"
    placeholder=""
    rows="2"
  />
   
  <!---->
</div>
`;
